### PR TITLE
enhancement(http sink): add bearer auth strategy

### LINKED ISSUE #2599: Bearer authentication strategy support?
Currently only `basic` authentication strategy supported through config options.

There also a question, which value should be used if config have valid `auth` and `Authorization` header? Should be validation error be thrown if two options set at one time and have different values?

cc @grantisdale

### PR DESCRIPTION
**Pull request by lukesteensen**
Closes #2599

### CODE REVIEW

**Comment by fanatid** [comment]
`loki` docs also should be changed, because `util::http::Auth` used there: https://github.com/timberio/vector/blob/0a0f8d5b6478237d7402ca9e356c9ff4ea6f8497/src/sinks/loki.rs#L50
`clickhouse` have `auth` section too, but use `util::http2::Auth` =\

File: src/sinks/util/http.rs [hunk 1]
```diff
@@ -346,16 +346,22 @@ impl RetryLogic for HttpRetryLogic {
...
+
         match &self {
             Auth::Basic { user, password } => {
-                use headers::HeaderMapExt;
                 let auth = headers::Authorization::basic(&user, &password);
                 req.headers_mut().typed_insert(auth);
             }
+            Auth::Bearer { token } => match headers::Authorization::bearer(&token) {
+                Ok(auth) => req.headers_mut().typed_insert(auth),
+                Err(error) => error!(message = "invalid bearer token", %token, %error),
```
Comment (LucioFranco) [inline src/sinks/util/http.rs:363]:
```suggestion
                Err(error) => error!(message = "invalid bearer token.", %token, %error),
```

**Review by LucioFranco — APPROVED** [review]

**Comment by LucioFranco** [comment]
Yeah, @lukesteensen we probably want to back port that to http2 or merge them. I don't think it matters too much with method we go with.

**Comment by fanatid** [comment]
Sorry I do not know code very good, but why there a lot of dependencies with different versions? (like `headers` crate here). Update process is hard? Or some dependecies use old crates?

**Comment by lukesteensen** [comment]
@fanatid yeah, we're in the process of updating a lot of our dependencies, so they're temporarily duplicated while we work through all the code that uses them.

File: src/sinks/util/http.rs (hunk 1 above)
Comment (fanatid) [reply src/sinks/util/http.rs:363]: I'm not sure, but maybe it's worth not log token value? Because there can be sensitive information.

Comment (lukesteensen) [reply src/sinks/util/http.rs:363]: Good point, we could remove that from both places.

**Pull request merged**